RLTSquare/CustomEmailSender/Controller/Router.php

```diff
@@ -0,0 +1,196 @@
+     */
+    protected StateInterface $inlineTranslation;
+    /**
+     * @var ScopeConfigInterface
+     */
+    protected ScopeConfigInterface $scopeConfig;
+    /**
+     * @var BackendTemplate
+     */
+    protected BackendTemplate $emailTemplate;
+    /**
+     * @var ActionFactory
+     */
+    private ActionFactory $actionFactory;
+
+    /**
+     * Router constructor.
+     *
+     * @param ActionFactory $actionFactory
+     * @param PageFactory $pageFactory
+     * @param Logger $logger
+     * @param TransportBuilder $transportBuilder
+     * @param StoreManagerInterface $storeManager
+     * @param StateInterface $state
+     * @param ScopeConfigInterface $scopeConfig
+     * @param BackendTemplate $emailTemplate
+     */
+    public function __construct(
+        ActionFactory         $actionFactory,
+        PageFactory           $pageFactory,
+        Logger                $logger,
+        TransportBuilder      $transportBuilder,
+        StoreManagerInterface $storeManager,
+        StateInterface        $state,
+        ScopeConfigInterface  $scopeConfig,
+        BackendTemplate       $emailTemplate
+    ) {
+        $this->actionFactory = $actionFactory;
+        $this->pageFactory = $pageFactory;
+        $this->logger = $logger;
+        $this->transportBuilder = $transportBuilder;
+        $this->storeManager = $storeManager;
+        $this->inlineTranslation = $state;
+        $this->scopeConfig = $scopeConfig;
+        $this->emailTemplate = $emailTemplate;
+    }
+
+    /**
+     * @param RequestInterface $request
+     * @return ActionInterface|null
+     */
+    public function match(RequestInterface $request): ?ActionInterface
+    {
+        $identifier = trim($request->getPathInfo(), '/');
+
+        if (strpos($identifier, 'rltsquare') !== false) {
+            $request->setModuleName('custom_email');
+            $request->setControllerName('index');
+            $request->setActionName('index');
+            $request->setParams([
+                'first_param' => 'first_value',
+                'second_param' => 'second_value'
+            ]);
+
+            $isEnabled = $this->scopeConfig->getValue(
+                self::Custom_Email_Config_Path,
+                ScopeInterface::SCOPE_STORE,
+                $this->storeManager->getStore()->getId()
+            );
+
+            if ($isEnabled) {
+                echo "Test";
+                $this->sendEmail();
+                $this->logger->info('Page visited');
+                exit;
+            }
+            echo 'Custom Email Module is disabled. Please Enable Module from Admin Configuration.';
+            return $this->actionFactory->create(Forward::class);
+        }
+        return null;
+    }
+
+    /**
+     * @return void
+     * @throws NoSuchEntityException
+     */
+    public function sendEmail(): void
+    {
+        // template id
+        $email_template = $this->emailTemplate->load('test_email_template', 'orig_template_code');
```

Comment:
Overlooked logic - the logic here is faulty. The requirement was to email template that user can edit and change from admin. That implies use of system.xml, with field specific for juggling email_template, which in turn gets set to default value of coded email template using config.xml.

```
               <field id="email_template" translate="label comment" type="select" sortOrder="30" showInDefault="1" showInWebsite="1" showInStore="1" canRestore="1">
                    <label>Email Template</label>
                    <comment>Email template chosen based on theme fallback when "Default" option is selected.</comment>
                    <source_model>Magento\Config\Model\Config\Source\Email\Template</source_model>
                </field>
```

Then, a helper method can be used for reading the template

```
    public function getEmailTemplate(): string
    {
        return (string)$this->scopeConfig->getValue(
            self::XML_PATH_EMAIL_TEMPLATE,
            ScopeInterface::SCOPE_STORE
        );
    }
```

```
            $transport = $this->transportBuilder
                ->setTemplateIdentifier($this->helper->getEmailTemplate())
```

this way we ensure either default meail template is used (through cdefaults in config.xml) or a reassigned one is used ... all through that one helper call